### ISSUE TITLE
[naming-convention] Feature request: allow string array for matches

### ISSUE THREAD

**Issue by blordpluto** [typescript-eslint/typescript-eslint#3141]
Feature request: This ignore syntax:
```
      "filter": {
        // you can expand this regex to add more allowed names
        "regex": "^(Property-Name-One|Property-Name-Two)$",
        "match": false
      }
```
becomes cumbersome when the list of allowed names gets even a little bit long. It works, but it quickly overflows a 100-character window. Diffs get harder to read, etc. A simple string array would be more ergonomic DX in many cases, because linebreaks.

Awesome tool here. Thanks for maintaining.  👍

**Comment by bradzacher** [comment]
If you're using a js config (which you should, as json configs are on their way out), you can do this in the config pretty easily.
```js
config = {
  filter: {
    regex: `^(${['Property-Name-One', 'Property-Name-Two'],join('|')})$`,
    match: false,
  }
}
```

The problem is that things can get a bit ambiguous if we allow an array.
For example, imagine that you provide:
```
filter: [
  { regex: '^Foo$', match: false },
  { regex: '^Foo$', match: true },
]
```
What would that mean? Those two filters directly contradict - meaning you can easily get yourself in an impossible-to-handle state.

**Comment by blordpluto** [comment]
haha yes thanks, forgot about dynamic operations in RC files. Stuck in an old mindset I guess. Fixed, thanks!
